config/management/src/validator_config.rs

```diff
@@ -1,202 +1,127 @@
+        validator_address: NetworkAddress,
+        reconfigure: bool,
+    ) -> Result<Transaction, Error> {
+        let mut storage = StorageWrapper::new(
+            self.validator_backend.name(),
+            &self.validator_backend.validator_backend,
+        )?;
+        let owner_account = storage.account_address(OWNER_ACCOUNT)?;
 
-// Decodes adn decrypts the validator address
-pub fn decode_validator_address(
-    address: RawEncNetworkAddress,
-    account: &AccountAddress,
-    key: &Key,
-    addr_idx: u32,
-) -> Result<NetworkAddress, Error> {
-    let enc_addr = EncNetworkAddress::try_from(&address).map_err(|e| {
-        Error::UnexpectedError(format!(
-            "Failed to decode network address {}",
-            e.to_string()
-        ))
-    })?;
-    let raw_addr = enc_addr.decrypt(key, account, addr_idx).map_err(|e| {
-        Error::UnexpectedError(format!(
-            "Failed to decrypt network address {}",
-            e.to_string()
-        ))
-    })?;
-    NetworkAddress::try_from(&raw_addr).map_err(|e| {
-        Error::UnexpectedError(format!(
-            "Failed to decode network address {}",
-            e.to_string()
-        ))
-    })
+        let consensus_key = storage.ed25519_public_from_private(CONSENSUS_KEY)?;
+        let fullnode_network_key = storage.x25519_public_from_private(FULLNODE_NETWORK_KEY)?;
+        let validator_network_key = storage.x25519_public_from_private(VALIDATOR_NETWORK_KEY)?;
+
+        // Build Validator address including protocols and encryption
+        // Append ln-noise-ik and ln-handshake protocols to base network addresses
+        // and encrypt the validator address.
+        let validator_address =
+            validator_address.append_prod_protos(validator_network_key, HANDSHAKE_VERSION);
+        let raw_validator_address = encode_address(validator_address)?;
+        // Only supports one address for now
+        let addr_idx = 0u32;
+        let key = TEST_SHARED_VAL_NETADDR_KEY;
+        let version = TEST_SHARED_VAL_NETADDR_KEY_VERSION;
+        let enc_validator_address = raw_validator_address.encrypt(
+            &key,
+            version,
+            &owner_account,
+            sequence_number + 1,
+            addr_idx,
+        );
+        let raw_enc_validator_address = RawEncNetworkAddress::try_from(&enc_validator_address)
+            .map_err(|e| {
+                Error::UnexpectedError(format!(
+                    "error serializing encrypted address: '{:?}', error: {}",
+                    enc_validator_address, e
+                ))
+            })?;
+
+        // Build Fullndoe address including protocols
+        let fullnode_address =
+            fullnode_address.append_prod_protos(fullnode_network_key, HANDSHAKE_VERSION);
+        let raw_fullnode_address = encode_address(fullnode_address)?;
+
+        // Generate the validator config script
+        // TODO(philiphayes): remove network identity pubkey field from struct
```

Comment:
What's this all about btw?  I copied it, then you copied it.

Reply:
the *network_keys below need to be removed from various configs, scripts, and smart contracts, I pinged @phlip9 to try to get it out in this cut, since this is a breaking change.